beacon-chain/state/stategen/getter.go

```diff
@@ -23,6 +23,11 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (*state.Bea
 		return s.beaconDB.State(ctx, blockRoot)
 	}
 
+	// Short cut if the state is already in the DB.
```

Comment:
Should you check if there is a cache hit before reading from the DB? It could be an even faster short cut.